### ISSUE TITLE
Mac OS Stream Deck Software?

### ISSUE THREAD

**Issue by simcc** [Zayik/CommandSender#2]
Does this only work with Windows stream deck software?

**Comment by Zayik** [comment]
Hey simcc,

The Elgato Stream Deck has software built for Windows and Mac OS.

I believe the plugin should function just fine on Mac OS as well.
Unfortunately, I do not have any way of testing that myself as I don't own
a mac.

However, I would be more than happy to troubleshoot with you should you run
into any issues.


Here is an example of a program you would pair with the command sender.
Zayik/UDPListener: Starts a basic UDP Listener that user can use to test
with a UDP Sender. (github.com) <https://github.com/Zayik/UDPListener>

Since the command sender is just sending a message via udp protocol, you
can technically write a receiver in any language (C#, Java, Python, C++,
etc.) and on any platform.
For example, one could create a program on a Raspberry Pi, and when it
receives a UDP message, it responds back with the current time.

**Comment by Zayik** [comment]
As I have heard no comments to suggest the plugin does not work on other platforms, I am going to mark this as closed.

**Comment by simcc** [comment]
Hi sorry for delay getting back. Thanks for the reply. I'm actually fine with the receivers etc. My problem is it won't even install on OSX. See Screenshot.
<img width="555" alt="Screenshot 2021-12-30 at 11 51 17" src="https://user-images.githubusercontent.com/42849565/147749683-c1a0b22e-34f9-4be7-a3ef-058d830dcb42.png">
.

**Comment by simcc** [comment]
I've only just started with stream deck, so no idea why it won't install...some plugins don't work on Mac and vice versa, no idea why...there's a brief list here (obvious reasons that some don't...but not all): https://streamdeck-plugins.com

**Comment by Zayik** [comment]
I've done some research, and I found out that depending on the coding language and features used, Mac OS isn't supported. 
This was done using a library to allow use of c#, which has Windows requirement. 

Therefore, this app unfortunately does not support Mac OS. 

That being said, I am looking at converting this to Javascript which can more easily support Mac OS. I'll let you know when I have that available.

**Comment by Zayik** [comment]
I have created a new plugin which should have Mac OS support. https://github.com/Zayik/CommandSender-Websocket-Edition
This is by switching to Javascript which Streamdeck indicates should be able to have Mac OS support.

However, there are some key differences. Instead of using UDP messages, it is using WebSockets. 
To the best of my knowledge, UDP messages cannot be used in Javascript unless used within Node JS, or possible some other technology. To the best of my knowledge, StreamDeck does not support any technologies that would allow use of udp messages within Javascript. This is why I switched over to WebSockets. 

To create a test server, or to learn how to interact with Websockets from a language such as C#, one can refer to https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_server. Mozilla provides a very simplistic example that can be compiled and used to test the new WebSocket version of the CommandSender.
